[FIX] web, website: fix the values set on the buttons CSS variables

sobo-odoo opened the pull request:
In commit [1], the "Themes Options" have been reorganized to make them
easier to find. With this relayout, the buttons style options have also
been modified in order to have a third choice "Flat", in addition to
"Fill" and "Outline". However, the way it was done causes some issues.

Indeed, it wrongly sets the `btn-(primary|secondary)-(outline|flat)` CSS
variables to `'True'` instead of the `true` boolean, and to `null`
instead of `false`, which is not correct. This happens because
`customizeButtonStyle` now sends boolean values to the `_makeSCSSCusto`
function, instead of strings representing these booleans. This results
in the backend RPC call setting the variables to the Python boolean
`True`, or to `null` in the falsy case because of the `"null"` default
value.

While it does not seem to break anything in general, there are issues
when using themes that redefine these variables (e.g. "Avantgarde",
"Enark", "Cobalt",...). Indeed, it is impossible to set the buttons
corresponding style to "Fill". This happens because choosing "Fill" is
supposed to set both variables to `false`, but because of the boolean
values being sent to `_makeSCSSCusto`, the value that is in fact set is
`"null"`. This therefore makes them fallback to the value defined in the
theme, forcing them to `true` and preventing the "Fill" style from being
applied.

This commit fixes these issues by properly giving string values to
`_makeSCSSCusto`. For already existing databases, all the variables that
have been set to `'True'` will have their value replaced by `true`,
thanks to the `o-map-omit` SCSS function. Note that the `'False'` case
is also added, in order to be consistent.

[1]: https://github.com/odoo/odoo/commit/388e4bb2bfcaebdd4ff30277fb49a034592d7086

opw-3957157

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/170015.png)](https://mergebot.odoo.com/odoo/odoo/pull/170015)

sobo-odoo commented:
@qsm-odoo ready for review :slightly_smiling_face:

qsm-odoo commented on addons/web/static/src/legacy/scss/utils.scss:
```diff
@@ -221,10 +221,14 @@
 }
 
-// Function to remove all null values of a map
+// Function to remove all null values of a map. Also transforms `'True'` values
+// into a boolean `true`.
```
I'd say let's be a bit more consistent and handle "False" too (even if it was not part of the bug here). Also add a slight explanation about why this is done in this function. You can also add this to justify it more:
```scss
// TODO in master, this should be moved in website, as the only use case is in website user values files
```

qsm-odoo commented on addons/web/static/src/legacy/scss/utils.scss:
```diff
@@ -221,10 +221,14 @@
     @return $-with-support-font;
```
Commit message:
>  This, combined with the above issue, made it impossible to set
the buttons primary style to "Fill".

Nitpicking but I think this is not accurate, choosing a value (e.g. setting to "Fill") is buggy in the same way with or without the bug in the theme enark definition.

qsm-odoo reviewed (changes requested):
Thanks for the fix, some nitpicking :pray:

robodoo commented:
@sobo-odoo the target branch 'saas-16.4' has been disabled, you may want to close this PR.

robodoo commented:
This PR targets the disabled branch odoo/odoo:saas-16.4, it needs to be retargeted before it can be merged.

sobo-odoo commented:
@qsm-odoo I made the changes :slightly_smiling_face: I added the `'False'` case, added a better explanation in the commit message and rebased on 17.0 :wink:

qsm-odoo reviewed (approved):
As discussed, not entirely stable for potential custo using the `o-map-omit` function but as indicated, it is only currently used in website and I doubt any custo is using it. And even if they do, this change is unlikely to break something. And even if they do, it really should not be critical. Let's not overthink for once :grimacing: 

Thanks for the fix :+1: 

@robodoo r+
